packages/tradershub/src/features/cfd/flows/MT5/AvailableMT5AccountsList/AvailableMT5AccountsList.tsx

```diff
@@ -1,19 +1,30 @@
 import React from 'react';
+import { useAuthorize } from '@deriv/api';
+import { Provider } from '@deriv/library';
 import {
     TradingAccountCard,
     TradingAccountCardContent,
     TradingAccountCardLightButton,
 } from '../../../../../components';
 import { THooks } from '../../../../../types';
 import { MarketTypeDetails } from '../../../constants';
+import { JurisdictionModal } from '../../../modals/JurisdictionModal';
 import { MT5AccountIcon } from '../MT5AccountIcon';
 
 const AvailableMT5AccountsList = ({ account }: { account: THooks.MT5AccountsList }) => {
     const { description, title } = MarketTypeDetails[account.market_type || 'all'];
+    const { data: activeAccount } = useAuthorize();
+    const { setCfdState } = Provider.useCFDContext();
+    const { show } = Provider.useModal();
+
+    const trailingButtonClick = () => {
+        setCfdState('marketType', account.market_type);
+        !activeAccount?.is_virtual && show(<JurisdictionModal />); /* show MT5PasswordModal for demo */
+    };
 
     const LeadingIcon = () => <MT5AccountIcon account={account} />;
 
-    const TrailingButton = () => <TradingAccountCardLightButton />;
+    const TrailingButton = () => <TradingAccountCardLightButton onSubmit={trailingButtonClick} />;
```

Comment:
Do we really need this `TrailingButton` to be created here, can be used in return JSX directly tho

Reply:
It's was already refactored in other PR.